app/certificates.go

```diff
@@ -404,39 +430,73 @@ func NewCertificatesCommand() (CommandOut, error) {
 							CaPrivateKeyPEM: caPrivKey,
 						})
 						if err != nil {
-							return fmt.Errorf("failed to generate certificate: %s", err)
+							return fmt.Errorf("failed to generate certificate: %w", err)
 						}
-						yes, err := ConfirmPrompt(
+						return writeCertificates(
 							ctx,
-							fmt.Sprintf("storing the certificate private key at %s, do not share this key with anyone. confirm:",
-								ctx.Path("private-key-file"),
-							),
-						)
-						if err != nil || !yes {
-							return nil
-						}
-						err = ioutil.WriteFile(
-							ctx.Path("certificate-file"),
+							"client certificate",
 							certPem,
-							0600,
-						)
-						if err != nil {
-							return fmt.Errorf("failed to write ca certificate: %s", err)
-
-						}
-						err = ioutil.WriteFile(
-							ctx.Path("private-key-file"),
 							certPrivKey,
-							0600,
+							ctx.Path("certificate-file"),
+							ctx.Path("key-file"),
 						)
-						if err != nil {
-							return fmt.Errorf("failed to write ca private key: %s", err)
-						}
-
-						return nil
 					},
 				},
 			},
 		},
 	}, nil
 }
+
+func checkPath(ctx *cli.Context, path string) (bool, error) {
+	if fi, err := os.Stat(path); !errors.Is(err, os.ErrNotExist) {
+		// the file exists,
+		switch mode := fi.Mode(); {
+		case mode.IsRegular():
+			yes, err := ConfirmPrompt(
+				ctx,
+				fmt.Sprintf("file already exists at path %s, do you want to overwrite:", path),
+			)
+			if err != nil {
+				return false, fmt.Errorf("failed to confirm: %w", err)
+			}
+			return yes, nil
+		case mode.IsDir():
+			return false, fmt.Errorf("path cannot be a directory: %s ", path)
+		default:
+			return false, fmt.Errorf("invalid file path: %s (file mode=%s)", path, mode.String())
+		}
+	}
+	return true, nil
+}
+
+func writeCertificates(ctx *cli.Context, typ string, cert, key []byte, certPath, keyPath string) error {
+	if cont, err := checkPath(ctx, certPath); err != nil || !cont {
+		return err
+	}
+	if cont, err := checkPath(ctx, keyPath); err != nil || !cont {
+		return err
+	}
+
+	yes, err := ConfirmPrompt(
+		ctx,
+		fmt.Sprintf("storing the %s (private) key at %s, do not share this key with anyone. confirm:", typ, keyPath),
+	)
+	if err != nil {
+		return fmt.Errorf("failed to confirm: %w", err)
+	}
+	if !yes {
+		return nil
+	}
+	err = ioutil.WriteFile(certPath, cert, 0644)
+	if err != nil {
+		return fmt.Errorf("failed to write client certificate: %w", err)
```

Comment:
is this same function used for both the CA and client certificate? If so, I'd remove the word "client" from these

Reply:
maybe format with "typ" ?